src/test/java/part1/exercise/StreamsExercise2.java

```diff
@@ -74,7 +74,10 @@ public void employersStuffLists() {
     @Test
     public void indexByFirstEmployer() {
         Map<String, List<Person>> employeesIndex = getEmployees().stream()
-                .map(e -> new PersonEmployerPair(e.getPerson(), e.getJobHistory().get(0).getEmployer()))
+                .map(e -> e.getJobHistory().stream().findFirst()

```

Comment:
Use `flatMap(...findFirst().map(...).map(Streams::of).orElseGet(Streams::apply))` or `flatMap(...limit(1))`